cmake/CeleritasLibrary.cmake

```diff
@@ -173,110 +175,138 @@ function(celeritas_add_library target)
   # I.e. this should really be done at generation time.
   # So in the meantime we use CELERITAS_USE_VecGeom as a proxy.
 
-  if(NOT CELERITAS_USE_VecGeom OR NOT CELERITAS_USE_CUDA OR NOT _contains_cuda)
+  if(NOT CELERITAS_USE_VecGeom OR NOT CELERITAS_USE_CUDA OR NOT _cuda_sources)
+    if(CELERITAS_USE_HIP AND _cuda_sources)
+      # When building Celeritas libraries, we put HIP/CUDA files in shared .cu
+      # suffixed files. Override the language if using HIP.
+      set_source_files_properties(
+        ${_cuda_sources}
+        PROPERTIES LANGUAGE HIP
+      )
+    endif()
+
     add_library(${target} ${ARGN})
+    if(_all_props)
+      set_target_properties(${target} PROPERTIES ${_all_props})
+    endif()
+    if(PROJECT_NAME STREQUAL "Celeritas")
+      add_library(Celeritas::${target} ALIAS ${target})
+      install(TARGETS ${target}
+        EXPORT celeritas-targets
+        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        COMPONENT runtime
+      )
+    endif()
     return()
   endif()
 
   cmake_parse_arguments(_ADDLIB_PARSE
-    "STATIC;SHARED;MODULE"
+    "STATIC;SHARED;MODULE;OBJECT"
     ""
     ""
     ${ARGN}
   )
   set(_lib_requested_type "SHARED")
   set(_cudaruntime_requested_type "Shared")
-  set(__static_build FALSE)
-  if((NOT BUILD_SHARED_LIBS AND NOT _ADDLIB_PARSE_SHARED) OR _ADDLIB_PARSE_STATIC)
+  set(_staticsuf "_static")
```

Comment:
Since `_staticsuf` actually *does* change as a variable (depending on whether the library is static or not), but the others are constants, I ask you to reconsider hardcoding the names for now: we can restore them to variability later.

Reply:
> Since _staticsuf actually does change as a variable 

Then technically we should add one more variable (`_staticsuf_config`) (and actually (`_midsuf_config`) :)  ) used as the initial value.

Besides this, I still think that keeping the code flexible does not hurt that much, avoid having to reintroduce code later if we want to add the flexibility and removing the `_midsuf` marker actually make the code (somewhat) more confusing as it remove the semantic difference between `${target}` and `${target}${_midsuf}`.

Reply:
Keeping it flexible means adding more variables that may not ever be used. Can we add flexibility only when it's needed if the extra complexity can be postponed without consequence?